x-pack/plugins/apm/common/agent_name.ts

```diff
@@ -58,12 +59,20 @@ export const RUM_AGENT_NAMES: AgentName[] = [
   'opentelemetry/webjs',
 ];
 
+export const MOBILE_AGENT_NAMES: AgentName[] = ['iOS/swift', 'android/java'];
+
 export function isRumAgentName(
   agentName?: string
 ): agentName is 'js-base' | 'rum-js' | 'opentelemetry/webjs' {
   return RUM_AGENT_NAMES.includes(agentName! as AgentName);
 }
 
+export function isMobileAgentName(
```

Comment:
This doesn't seem to be used

Reply:
Thanks @gbamparop !

I pushed a commit now in which I replaced the usage of isOSAgentName with isMobileAgentName in all the places in the APM UI. Since the Android agent should be treated in the same way as the iOS agent.

Reply:
Thanks for updating this! 

> * is the method `normalizeAgentName` used anywhere? I couldn't find anything
It seems to have been introduced by https://github.com/elastic/kibana/pull/102346 but it's not used

> * why did we define `opentelemetry/swift`
It makes sense, maybe we hadn't considered that at the time (see https://github.com/elastic/kibana/pull/103485#discussion_r660469765)

> * Why do we [hide the dependencies view](https://github.com/elastic/kibana/blob/b72601c283bd9982ef2ed415b40e3cef179e19ac/x-pack/plugins/apm/public/components/routing/templates/apm_service_template/index.tsx#L224) for the `iOSAgent` / mobile agent?

It was raised as an issue https://github.com/elastic/kibana/issues/108510

Reply:
> > - is the method `normalizeAgentName` used anywhere? I couldn't find anything
>
> It seems to have been introduced by https://github.com/elastic/kibana/pull/102346 but it's not used

OK, I deleted it and simplified the code there.

> > - why did we define `opentelemetry/swift`
>
> It makes sense, maybe we hadn't considered that at the time

:+1: I reverted it to be just `iOS/swift` / `ios/swift`

> > - Why do we [hide the dependencies view](https://github.com/elastic/kibana/blob/b72601c283bd9982ef2ed415b40e3cef179e19ac/x-pack/plugins/apm/public/components/routing/templates/apm_service_template/index.tsx#L224) for the `iOSAgent` / mobile agent?
>
> It was raised as an issue https://github.com/elastic/kibana/issues/108510

I'll keep it as is for now and continue the discussion on this aspect in a separate thread.

Reply:
Thanks, do you think there's another field that we could use to find if an app instrumented with `opentelemetry/swift` is on iOS?